### PR TITLE
4204 i start use writing libraly csvhelper to convert xml to csv

### PR DESCRIPTION
**Pull request by Fai-nan**
I start learn how to  writing libraly csvhelper to convert xml to csv

### CODE REVIEW

File: ConvertCSVToXmlOrchard/ConvertCSVToXmlOrchard/ConvertCSVToXmlOrchard.csproj [hunk 1]
```diff
@@ -72,7 +72,6 @@
       <AutoGen>True</AutoGen>
       <DependentUpon>Resources.resx</DependentUpon>
     </Compile>
-    <None Include="bin\Debug\ConvertCSVToXmlOrchard.exe.config" />
```
Comment (jean) [inline ConvertCSVToXmlOrchard/ConvertCSVToXmlOrchard/ConvertCSVToXmlOrchard.csproj:75]: Cool :+1: 
Getting rid of build artifacts.

File: ConvertCSVToXmlOrchard/ConvertCSVToXmlOrchard/CsvToXml.cs [hunk 2]
```diff
@@ -64,20 +64,22 @@ private static void SaveToDataBase(XElement content)
...
                     ValueFromDatabase.Save(databasepath);
                 }
 
             }
             else
             {
-                var ValueFromDatabase = XElement.Load(listOfValueFromDatabase[0]); 
+                XDocument ValueFromDatabase = new XDocument();
+                ValueFromDatabase = new XDocument(new XDeclaration("1.0", "UTF-8", "yes"));
+                ValueFromDatabase.Add(new XComment("Exported from Orchard"),XElement.Parse(listOfValueFromDatabase[0]));
```
Comment (jean) [inline ConvertCSVToXmlOrchard/ConvertCSVToXmlOrchard/CsvToXml.cs]: That is not really true ;-)
This XML file is "Created by Fai", not "Exported from Orchard" :grimacing:

File: ConvertCSVToXmlOrchard/ConvertXmlToCsv/wf_xmlconvert.cs [hunk 3]
```diff
@@ -35,66 +35,92 @@ private void OpenFilexml()
...
+
+                    header = String.Join(",", name);
 
-        }
 
-        //public static List<Dictionary<string, string>>WritingInCSV(string absolutePath)
-        //{
+                    var query =
+                        from dealer in xml.Descendants("Content").Elements()
+                        select string.Join(",",
```
Comment (jean) [inline ConvertCSVToXmlOrchard/ConvertXmlToCsv/wf_xmlconvert.cs]: Hmm, here you are creating CSV lines, like `"value1","value2","value3"`?
This is what `CsvHelper` should do for you. 
You should be passing an enumerable of objects to `CsvHelper`, like `csv.WriteRecords( records );`, or if you aren't using a class, using `csv.WriteField(x)`. 

It looks like you want to do the writing twice, in [l106](https://github.com/salween/ToolsConvert/pull/4/files#diff-30f1d823aa8372641e7a88bdee2e9ca0R106) and again in [XmltoCsv l22](https://github.com/salween/ToolsConvert/pull/4/files#diff-c880c42188cd6e578dacf416c056180dR22), but I'm not sure I understand what is happening there. It's preparing a writer but not using it.

File: ConvertCSVToXmlOrchard/ConvertXmlToCsv/wf_xmlconvert.cs [hunk 4]
```diff
@@ -35,66 +35,92 @@ private void OpenFilexml()
...
+                    SaveFileDialog saveFileDialog1 = new SaveFileDialog();
+                    saveFileDialog1.Filter = "CSV files (*.csv)|*.csv";
+                    DialogResult dr = saveFileDialog1.ShowDialog();
+                    if (dr == DialogResult.OK)
+                    {
+                        string filename = saveFileDialog1.FileName;
+
+                        File.WriteAllText(filename, csv);
+
+                        XmltoCsv ss = new XmltoCsv();
```
Comment (jean) [inline ConvertCSVToXmlOrchard/ConvertXmlToCsv/wf_xmlconvert.cs]: What is `ss`?

File: ConvertCSVToXmlOrchard/ConvertCSVToXmlOrchard/CsvToXml.cs (hunk 2 above)
Comment (Fai-nan) [reply ConvertCSVToXmlOrchard/ConvertCSVToXmlOrchard/CsvToXml.cs]: OK

File: ConvertCSVToXmlOrchard/ConvertXmlToCsv/wf_xmlconvert.cs (hunk 3 above)
Comment (Fai-nan) [reply ConvertCSVToXmlOrchard/ConvertXmlToCsv/wf_xmlconvert.cs]: I still do not finish working  and and try to edit

Comment (jean) [reply ConvertCSVToXmlOrchard/ConvertXmlToCsv/wf_xmlconvert.cs]: :ok_hand:

File: ConvertCSVToXmlOrchard/ConvertCSVToXmlOrchard/CsvToXml.cs (hunk 2 above)
Comment (jean) [reply ConvertCSVToXmlOrchard/ConvertCSVToXmlOrchard/CsvToXml.cs]: @Fai-nan I added https://github.com/salween/ToolsConvert/pull/4/commits/6bebe36d2e5314a9328fdceb5c2732152fc77433 to address this.

File: ConvertCSVToXmlOrchard/ConvertCSVToXmlOrchard/CsvToXml.cs [hunk 5]
```diff
@@ -93,23 +95,23 @@ private static void LoadFromDataBase()
...
-                                   Phone = s.Element("TextField.Phone").Attribute("Text").Value,
-                                   Email = s.Element("TextField.Email").Attribute("Text").Value,
-                                   Webpage = s.Element("TextField.Webpage").Attribute("Text").Value,
-                                   Spouse = s.Element("TextField.Spouse").Attribute("Text").Value,
-                                   Guest = s.Element("TextField.Guest").Attribute("Text").Value,
-                                   Firsttime = s.Element("TextField.Firsttime").Attribute("Text").Value
-                               }).ToList();
+                                select new
+                                {
+                                    Iddentity = s.Element("IdentityPart").Attribute("Identifier").Value,
```
Comment (jean) [inline ConvertCSVToXmlOrchard/ConvertCSVToXmlOrchard/CsvToXml.cs]: Typo: `Iddentity` -> `Identity`

**Pull request merged**